minetest/minetest
# Fix multiple password changes in one session

Currently, if you try to change your password and then change it again, it won't work properly. The server won't accept the new password after the first change but will accept the old one like the change had no effect.
And if you try to change your password right after registration, the server won't even reply that the change was accepted or denied. This PR fixes both cases.

## To do

This PR is Ready for Review.

## How to test

1. Login by creating a new account.
2. Try to change the password with an incorrect password, see "Password change denied".
3. Change the password with the correct password, see "Password changed".
4. Try to change the password again with the old password, see "Password change denied".
5. Change the password with the new password, see "Password changed".
6. Logout and login again with the latest password.
